### PR TITLE
Geometry_Engine: Minor method addition

### LINKED ISSUE #1616: Geometry_Engine: Missing methods for PlanarSurface
#### Broken rules:
All `I` methods that take an `IGeometry` or an `ISurface` as first input should have a corresponding method for all possible geometry/surface. This is not the case for `PlanarSurface`.

(This was highlighted in https://github.com/BHoM/Grasshopper_Toolkit/issues/489)

#### Suggestions to restore compliance:
Implement all missing geometry methods for PlanarSurface.

### Comments:
- @FraserGreenroyd, @awakeman , this is/will be part of the CI check you are doing on compiled code. So I'm flagging you up here in case you are already able to provide the full list of missing methods (potentially for more than PlanarSurface).
- Assigning @IsakNaslundBh and @pawelbaran since they were the ones assigned for the creation of the PlanarSurface (see https://github.com/BHoM/BHoM/issues/424) but feel free to reassign appropriately.

### LINKED ISSUE #1771: Geometry_Engine: Add vector from two points method

### LINKED ISSUE #1772: Geometry_Engine: Add IIsNurbsCurve to check if any part of a curve is a nurbs
#### Description:

image

### LINKED ISSUE #1878: Geometry_Engine: implement Flip for NurbsCurve
#### Description:

I've experienced a few issues caused by that one missing. Please correct me if I am wrong, shouldn't it be enough to reverse the order of control points, weights and the knot vector?

### LINKED ISSUE #1879: Geometry_Engine: IsEqual for Basis
#### Description:

Needed for ETABS checks

### PR DESCRIPTION
**Pull request by kThorsager**
### Issues addressed by this PR

Closes #1771 
Closes #1772 
Closes #1878 
Closes #1616 
Closes #1879 

Added a bunch of small requested methods

### Test files

https://burohappold.sharepoint.com/:u:/s/BHoM/EcsuiThG1spMufdZ7gsofIgB7vijPXV-FYZtBkBcNtwenA?e=e6OxIq

### Changelog

- Added Create Vector from two Points
- Added IsNurbsCurve
- Implemented: Clone, ClosestPoint, IsEqual and IsInPlane for PlanarSurface
- Added IsEqual for Basis
- Implemented Flip for NurbsCurve

### CODE REVIEW

File: Geometry_Engine/Query/IsNurbsCurve.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        /***************************************************/
+
+        [Description("Queries if any part of the curve is a nurbs curve.")]
+        [Input("curve", "Curve to query.")]
+        [Output("true if any part of the curve is a nurbs curve.")]
+        public static bool IsNurbsCurve(this ICurve curve)
+        {
+            return curve.ISubParts().Any(x => x is NurbsCurve); ;
+        }
+
```
Comment (pawelbaran) [inline Geometry_Engine/Query/IsNurbsCurve.cs:45]: `/***************************************************/` missing 🤖

**Comment by kThorsager** [comment]
> No support for single points in `IsContaining` keeps on bothering me (each time it is called in the code, a single point gets wrapped into a list), but not sure if that is something we could make any better (apart from adding yet another set of methods taking a single `Point`). Anyways, that is out of the scope of this PR, just my personal reflection.

hm, that is annoying, suspect adding a wrapping method which just puts the single point in a list and call the method. As some things are pre-processed,, so flipping is isn't great

**Comment by pawelbaran** [comment]
100% agreed @kThorsager. Then again, not sure if the wrapper method will not cause more confusion than comfort (more methods with same name etc.). What do you think?

**Comment by kThorsager** [comment]
It will be more methods, but it would be enough to add it to the interface one. And the do the same thing, so confusion should be minimal.

**Comment by pawelbaran** [comment]
In such case, I would vote for adding one and replacing all calls taking a list with single points with the actual points. However, very low prio for sure.

File: Geometry_Engine/Query/Clone.cs [hunk 2]
```diff
@@ -136,6 +136,13 @@ public static NurbsSurface Clone(this NurbsSurface surface)
 
         /***************************************************/
 
+        public static PlanarSurface Clone(this PlanarSurface surface)
+        {
+            return new PlanarSurface(surface.ExternalBoundary, surface.InternalBoundaries.ToList());
```
Comment (IsakNaslundBh) [inline Geometry_Engine/Query/Clone.cs:141]: Think the boundaries, internal and external, should be cloned here as well. That has at least been the general approach for the rest of the close methods.

Comment (kThorsager) [reply Geometry_Engine/Query/Clone.cs:141]: Oh, yeah, that makes sense

**Pull request merged**